zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryResponseMapper.java

```diff
@@ -7,43 +7,66 @@
  */
 package io.camunda.zeebe.gateway.rest;
 
+import static java.util.Optional.ofNullable;
+
+import io.camunda.service.entities.DecisionDefinitionEntity;
 import io.camunda.service.entities.ProcessInstanceEntity;
 import io.camunda.service.search.query.SearchQueryResult;
+import io.camunda.zeebe.gateway.protocol.rest.DecisionDefinitionItem;
+import io.camunda.zeebe.gateway.protocol.rest.DecisionDefinitionSearchQueryResponse;
 import io.camunda.zeebe.gateway.protocol.rest.ProcessInstanceItem;
 import io.camunda.zeebe.gateway.protocol.rest.ProcessInstanceSearchQueryResponse;
 import io.camunda.zeebe.gateway.protocol.rest.SearchQueryPageResponse;
 import io.camunda.zeebe.util.Either;
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 import org.springframework.http.ProblemDetail;
 
 public final class SearchQueryResponseMapper {
 
-  public SearchQueryResponseMapper() {}
+  private SearchQueryResponseMapper() {}
 
   public static Either<ProblemDetail, ProcessInstanceSearchQueryResponse>
       toProcessInstanceSearchQueryResponse(final SearchQueryResult<ProcessInstanceEntity> result) {
-    final var response = new ProcessInstanceSearchQueryResponse();
-    final var total = result.total();
-    final var sortValues = result.sortValues();
-    final var items = result.items();
-
-    final var page = new SearchQueryPageResponse();
-    page.setTotalItems(total);
-    response.setPage(page);
+    final var page = toSearchQueryPageResponse(result);
+    final var response =
+        new ProcessInstanceSearchQueryResponse()
+            .page(page)
+            .items(
+                ofNullable(result.items())
+                    .map(SearchQueryResponseMapper::toProcessInstances)
+                    .map(Either::get)
+                    .orElseGet(ArrayList::new));
```

Comment:
🙃 Just a nitpick but I guess we could also use the `Collections::emptyList` here instead of creating a new list instance? It's just a small thing though. There are more code lines like this in this PR.

Reply:
That's correct. The only "drawback" of using `Collections.emptyList `is its immutability. If someone tries to add an item to that list, it will cause an error. However, such an attempt would indicate a code smell. 
I'll make the change.

Reply:
✅